src/satosa/frontends/openid_connect.py

```diff
@@ -256,7 +256,13 @@ def _get_approved_attributes(self, provider_supported_claims, authn_req):
         if "claims" in authn_req:
             for k in ["id_token", "userinfo"]:
                 if k in authn_req["claims"]:
-                    requested_claims.extend(authn_req["claims"][k].keys())
```

Comment:
@mrvanes I'm going through this PR, but I'm sure how this is useful. This is some test data for `authn_req["claims"]`:

```python
authn_req["claims"] = {
  "id_token": {
    "foo": "bar",
    "mail": {
      "essential": True
    }
  },
  "userinfo": {
    "acr": {
      "essential": True,
      "values": [
        "urn:mace:incommon:iap:silver",
        "urn:mace:incommon:iap:bronze"
      ]
    }
  }
}
```

Coming out of the process, the result is:
```python
> print(requested_claims)
['sub', 'foo', 'email.essential', 'acr.essential', 'acr.values']
```

Do we really want `email.essential`? How is it more correct than before?